test-app/tests/integration/components/multiselect-test.gts

```diff
@@ -0,0 +1,1192 @@
+
+    await render(<template>
+      <Multiselect
+        @options={{testColors}}
+        @selected={{selected}}
+        @removeButtonLabelFunction={{formatRemoveLabel}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option data-option>
+          {{multiselect.option}}
+        </multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    assert.dom('[data-multiselect-remove-option]').exists({ count: 2 });
+
+    let chips = document.querySelectorAll('[data-multiselect-remove-option]');
+
+    assert.dom(chips[0]).hasAttribute('aria-label', 'Remove "blue"');
+    assert.dom(chips[1]).hasAttribute('aria-label', 'Remove "red"');
+  });
+
+  test('it sets `aria-expanded` based on the popover state', async function (assert) {
+    await render(<template>
+      <Multiselect
+        @options={{testColors}}
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option data-option>
+          {{multiselect.option}}
+        </multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    assert.dom('[role="listbox"]').doesNotExist();
+
+    assert.dom('[data-multiselect]').hasNoAttribute('aria-expanded');
+
+    await click('[data-multiselect]');
+
+    assert.dom('[data-multiselect]').hasAttribute('aria-expanded');
+  });
+
+  test('it sets `aria-controls`', async function (assert) {
+    await render(<template>
+      <Multiselect
+        @options={{testColors}}
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option data-option>
+          {{multiselect.option}}
+        </multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    assert.dom('[data-multiselect]').hasAttribute('aria-controls');
+  });
+
+  test('it applies the provided `@contentClass` to the popover content list', async function (assert) {
+    await render(<template>
+      <Multiselect
+        @options={{testColors}}
+        @contentClass="test-class"
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+      />
+    </template>);
+
+    await click('[data-multiselect]');
+
+    assert.dom('[role="listbox"]').exists();
+    assert.dom('[role="listbox"]').hasClass('test-class');
+  });
+
+  test('it renders the provided options in the popover list', async function (assert) {
+    await render(<template>
+      <Multiselect
+        @options={{testColors}}
+        @contentClass="test-class"
+        @removeButtonLabelFunction={{removeButtonLabelFunction}}
+        data-multiselect
+        as |multiselect|
+      >
+        <multiselect.Option data-option>
+          {{multiselect.option}}
+        </multiselect.Option>
+      </Multiselect>
+    </template>);
+
+    await click('[data-multiselect]');
+
+    assert.dom('[role="option"]').exists({ count: 2 });
```

Comment:
Seems a little odd reading through this that we test `role="option"` _and_ for `data-option`, but we want to ensure we have the `role` attribute set properly here.  Worth commenting about or is that clear?

Reply:
`data-option` is essentially ensuring we `...attributes` on each individual option, so since we have `<multiselect.Option data-option>`, we expect `data-option` to be on each option too.  Let me add a comment here for clarification!